### ISSUE TITLE
Spikes in training loss

### ISSUE THREAD

**Issue by slala2121** [google-research/simclr#189]
I'm wondering if you have encountered this recurring spike in the training loss and why might this occur. 

It does appear to occur upon processing the last batch but I don't think it has to do with uneven batch size because the last batch is dropped. The data is shuffled so it can't be any particular examples causing this. But it's not clear why the processing of the last batch would produce these spikes. 

Thanks.
 
![Screen Shot 2022-02-06 at 12 37 55 AM](https://user-images.githubusercontent.com/58089750/152669016-21771135-bce7-46cd-8c09-f4905f9e0976.png)

**Comment by thecooltechguy** [comment]
@slala2121 I'm actually experiencing the exact same behavior, and I'm also shuffling + dropping the last batch. Did you happen to figure out the reason or resolve this?
